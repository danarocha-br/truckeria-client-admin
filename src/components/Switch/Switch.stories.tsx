import { Story, Meta } from '@storybook/react/types-6-0';

import Switch from '.';

export default {
  title: 'Form Controls/Switch',
  component: Switch,
} as Meta;

export const ThemeSwitch: Story = () => <Switch />;
