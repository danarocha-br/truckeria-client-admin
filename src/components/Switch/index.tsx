import * as React from 'react';

import Switch from 'react-switch';

import tokens from '../../../tailwind.config';
import useDarkMode from 'use-dark-mode';

const SwitchCustom = () => {
  const darkMode = useDarkMode(true);

  return (
    <Switch
      onChange={darkMode.toggle}
      checked={darkMode.value}
      aria-label="Theme Switcher"
      checkedIcon={false}
      uncheckedIcon={false}
      handleDiameter={20}
      offColor={tokens.theme.colors.neutral[700]}
      onColor="#fff"
      offHandleColor={tokens.theme.colors.neutral[700]}
      onHandleColor="#fff"
      height={60}
      width={60.01}
      borderRadius={0}
      activeBoxShadow={'0 0 2px 3px #ffffff0'}
      uncheckedHandleIcon={
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          fill="none"
          stroke="#fff"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
        >
          <path d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z" />
        </svg>
      }
      checkedHandleIcon={
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          fill="none"
          stroke="currentColor"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
        >
          <circle cx="12" cy="12" r="5" />
          <path d="M12 1v2M12 21v2M4.22 4.22l1.42 1.42M18.36 18.36l1.42 1.42M1 12h2M21 12h2M4.22 19.78l1.42-1.42M18.36 5.64l1.42-1.42" />
        </svg>
      }
    />
  );
};

export default SwitchCustom;
