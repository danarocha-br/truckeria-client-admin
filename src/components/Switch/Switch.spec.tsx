import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Switch from '.';

describe('<Switch />', () => {
  it('should render the Switch', () => {
    const { container } = renderWithTheme(<Switch />);

    expect(
      screen.getByRole('switch', { name: /Theme Switcher/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
