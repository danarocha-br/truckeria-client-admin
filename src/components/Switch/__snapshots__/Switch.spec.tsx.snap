// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render the Switch 1`] = `
<div
  style="position: relative; display: inline-block; text-align: left; opacity: 1; direction: ltr; border-radius: 30px; transition: opacity 0.25s; user-select: none;"
>
  <div
    class="react-switch-bg"
    style="height: 60px; width: 60.01px; margin: 0px; position: relative; background: rgb(255, 255, 255); border-radius: 0; cursor: pointer; transition: background 0.25s;"
  />
  <div
    class="react-switch-handle"
    style="height: 20px; width: 20px; background: rgb(255, 255, 255); display: inline-block; cursor: pointer; border-radius: -1px; position: absolute; transform: translateX(20.009999999999998px); top: 20px; outline: 0; border: 0px; transition: background-color 0.25s, transform 0.25s, box-shadow 0.15s;"
  >
    <div
      style="height: 20px; width: 20px; opacity: 0; position: absolute; left: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
    >
      <svg
        fill="none"
        height="24"
        stroke="#fff"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 12.79A9 9 0 1111.21 3 7 7 0 0021 12.79z"
        />
      </svg>
    </div>
    <div
      style="height: 20px; width: 20px; opacity: 1; position: absolute; left: 0px; top: 0px; pointer-events: none; transition: opacity 0.25s;"
    >
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="5"
        />
        <path
          d="M12 1v2M12 21v2M4.22 4.22l1.42 1.42M18.36 18.36l1.42 1.42M1 12h2M21 12h2M4.22 19.78l1.42-1.42M18.36 5.64l1.42-1.42"
        />
      </svg>
    </div>
  </div>
  <input
    aria-checked="true"
    aria-label="Theme Switcher"
    checked=""
    role="switch"
    style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px;"
    type="checkbox"
  />
</div>
`;
